pkg/cli/tail.go

```diff
@@ -176,21 +200,50 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("-max-num must be greater than 0 and less than 1000")
 	}
 
-	// Tail logs.
-	ls, err := c.tail(ctx)
+	var extra []validation.Validator
+	if c.flagAdditionalCheck {
+		extra = append(extra, validation.ValidateLabels)
+	}
+
+	var puller logPuller
+	if c.testPuller != nil {
+		puller = c.testPuller
+	} else {
+		logClient, err := logging.NewClient(ctx)
+		defer func() {
+			if err := logClient.Close(); err != nil {
```

Comment:
I wonder if it would panic, if there is err when `logging.NewClient(ctx)` as it returns nil logClient.

Reply:
Updated the code to check if `logClient == nil` before calling `logClient.Close()`